crates/objc2/src/rc/id.rs

```diff
@@ -162,7 +162,7 @@ pub type Id<T> = Retained<T>;
 
 impl<T: ?Sized> Retained<T> {
     #[inline]
-    pub(crate) unsafe fn new_nonnull(ptr: NonNull<T>) -> Self {
+    pub unsafe fn new_nonnull(ptr: NonNull<T>) -> Self {
```

Comment:
I don't want to expose this, users should use `Retained::from_raw(ptr.as_ptr()).unwrap()` instead as the unwrap should be optimized away (and if it isn't, we have `Option::unwrap_unchecked` for that).

Reply:
Hmm that seems a bit silly when there is an API primitive available that is nicer and easier to read?

`from_raw()` uses this under the hood after all, and the only difference I see is the extra `+ Message` bound to get access to it.

Reply:
Agree that it is a bit silly; but I'm trying to follow `std` here, and the prime examples of smart pointers, `Box` and `Arc`, only expose [`Box::from_raw`](https://doc.rust-lang.org/nightly/std/boxed/struct.Box.html#method.from_raw) and [`Arc::from_raw`](https://doc.rust-lang.org/nightly/std/sync/struct.Arc.html#method.from_raw), and I don't really want to differ from these here if I can avoid it.

Reply:
I hadn't thought about matching `std` at all, where I don't use these functions very often (they're after all only to for raw pointers that originate from Rust at some point, i.e. temporarily creating a "user pointer" for FFI).

Here in `objc2` it seems to make sense given that your bindings already generate a `NonNull` annotation where applicable.